src/sentry/sentry_apps/token_exchange/refresher.py

```diff
@@ -78,18 +85,36 @@ def token(self) -> ApiToken:
         try:
             return ApiToken.objects.get(refresh_token=self.refresh_token)
         except ApiToken.DoesNotExist:
-            raise APIUnauthorized("Token does not exist")
+            raise SentryAppIntegratorError(
+                message="Given refresh token does not exist",
+                webhook_context={
+                    "token": self.refresh_token[:4],
+                    "installation_uuid": self.install.uuid,
+                },
+            )
```

Comment:
`APIUnauthorized` has status code 401

```suggestion
            raise SentryAppIntegratorError(
                message="Given refresh token does not exist",
                webhook_context={
                    "token": self.refresh_token[:4],
                    "installation_uuid": self.install.uuid,
                },
                status_code=401
            )
```